feat: initial docker-compose support

jniles opened the pull request:
Here, we have created an initial docker-compose file that links together MySQL, Redis, and builds BHIMA.  In order to get it all working, I needed to make the REDIS_HOST into an environmental variable.

There are still some things that needs to done:

1. We need to have a better way of building the MySQL database.  Currently, the user has to run `yarn build:docker` to create an initial database file to populate the database.
2. We should be using multi-stage builds to ensure that we only have build artifacts in a "build" image and then deploy the most minimal image one can think of.
3. We need documentation on how to deploy/run with `docker` & `docker-compose`.

Closes #3439.

----

To get this working, you will need to have both [`docker`](https://docs.docker.com/engine/install/) and [`docker-compose`](https://docs.docker.com/compose/install/) installed.  Please follow those instructions for your operating system.  I've tested this on Ubuntu 20.04.

Then, please execute the following commands in the root directory of the BHIMA repository.

1. Add the `REDIS_HOST` to your `.env` file.  You can set it to `127.0.0.1` for running locally.  Docker will automatically change this to the appropriate address based on the docker-compose file.
2. `yarn build:docker` - this command will concatenate all the database files into a single SQL file.
3. `docker-compose up --build` - this command will create the BHIMA application, bring up MySQL and Redis, and link them all together in their own internal network.

The above command will give you the logs from the containers.  You can add the `-d` flag to docker to detach if you want to run in the background.

Once the containers are all up and running, the application should be available on localhost on the `$PORT` you have in your `.env` file.

Voila!  You should have the BHIMA installation page available.

It is probably useful to read through [dockerizing a nodejs app](https://nodejs.org/en/docs/guides/nodejs-docker-webapp/) for context.

jniles committed:
chore: initial docker-compose support

Here, we have created an initial docker-compose file that links together
MySQL, Redis, and builds BHIMA.  In order to get it all working, I
needed to make the REDIS_HOST into an environmental variable.

There are still some things that needs to done:

1. We need to have a better way of building the MySQL database.
   Currently, the user has to run yarn build:docker to get it working.
2. We should be using multi-stage builds to ensure that we only have
   build artifacts in a "build" image and then deploy the most minimal
   image one can think of.
3. We need documentation on how to deploy/run with docker &
   docker-compose.

Partially addresses #3439.

jniles committed:
wip(docker): ensure mysql builds database

Ensures that the database is built by MySQL.

jmcameron commented on Dockerfile:
```diff
@@ -0,0 +1,33 @@
+
+COPY .env bin/
+
+# change directory to the bin diretory
```
Thanks for the documentation.  It is really important to document your Dockerfile to indicate what changes are happening outside the container vs inside the container!

jmcameron reviewed (commented):
I tested this in Ubuntu LTS 20.04 inside WSL on Windows 10 and it worked.  I had a minor problem with installing docker-composed (see https://github.com/IMA-WorldHealth/bhima/pull/6229#issuecomment-1004205177).  One I fixed that, running 'docker-compose up --build' worked (it took about 6 minutes).  Then I was able to access Bhima via web broser, but got the "Basic System Installation" setup page. 
- We need scripts and documentation on how to run Bhima with specific production site data (without doing the Base System Installation).
- Restarting the container was fast (10+ seconds)
- Stopping the container was difficult.  ^C stopped it partially, but I eventually had to suspend and manually kill the process.
- With default configurations, building a docker container for Bhima requires multiple GB of data saved to your local computer.  It is probably worthwhile adding notes about where those files are stored and how to prune/delete the docker containers and images involved.  On my system, it looks like it required over 4GB of docker data to create the docker images and containers.

jniles committed:
perf: download chromium as a apt package

We've moved the chromium download from the node_nodules to an OS-level
download.  This ensures that don't spend a long in the node_modules
section of the build.

jniles committed:
chore: add charset to mysql

Adds the charset to MySQL.

jmcameron reviewed (approved):
It looks good to me.   It ran okay and I was able to see the installation page in my browser.  However, I still had difficulty stopping the docker container.  I did not see the OHADA errors.   I'll attach my log for the run so you can take a look at it.  I noticed several warnings, but they seemed harmless.
[docker-jmc-2022-01-06.log](https://github.com/IMA-WorldHealth/bhima/files/7823201/docker-jmc-2022-01-06.log)

jniles committed:
chore: ensure proper character set for mysql

This commit ensures the server is running in UTF8mb4 character set.
